File: apps/todo_svc/src/models/todo.model.ts
import { User, UserWithRelations } from '@loopback/authentication-jwt';
import {belongsTo, Entity, model, property} from '@loopback/repository';

@model()
export class Todo extends Entity {
  @property({
    type: 'string',
    id: true,
    generated: false,
    mongodb: {dataType: 'ObjectID'},
  })
  id?: string;

  @property({
    type: 'string',
    required: true,
    jsonSchema: {
      maxLength: 200,
      errorMessage: 'name must be less than 200 characters',
    },
  })
  title: string;

  @property({
    type: 'string',
    jsonSchema: {
      maxLength: 500,
      errorMessage: 'name must be less than 500 characters',
    },
  })
  desc?: string;

  @property({
    type: 'boolean',
  })
  isComplete?: boolean;

  @property({
    type: 'date',
    defaultFn: "now",
  })
  createdAt?: Date;


  @belongsTo(() => User)
  userId: string;


  constructor(data?: Partial<Todo>) {
    super(data);
  }
}

export interface TodoRelations {
  // describe navigational properties here
  user?: UserWithRelations;
}

export type TodoWithRelations = Todo & TodoRelations;

File: internal/todo/transport/model/todo_model.go
package model

import (
	"manabieAssignment/internal/core/entity"
	"time"
)

type TodoModel struct {
	UserID  int64
	Name    string
	Content string
}

func (t TodoModel) ToDomainModel() entity.Todo {
	return entity.Todo{
		UserID:    t.UserID,
		Name:      t.Name,
		Content:   t.Content,
		CreatedAt: time.Now(),
	}
}
